### ISSUE TITLE
Support for flat icons

### ISSUE THREAD

**Issue by simov** [amio/badgen-service#10]
I see that you support flat badges in `badgen`, are they supported here? I'm not seeing anything about flat icons inside the docs.

**Comment by amio** [comment]
I'm planning to support flat badge, thinking use `flat.badgen.net` to serve flat style and `badgen.net` for classic style, should be cleaner than switching with query params.

**Comment by amio** [comment]
It's online now https://flat.badgen.net 🤗

**Comment by simov** [comment]
Nice! :metal:

**Comment by styfle** [comment]
Why .net instead of the .now.sh TLD?

**Comment by amio** [comment]
- Prefer shorter url
- Cleaner way to switch style (domain vs `?style=flat`)
- badgen.net & flat.badgen.net could serve a full preview of badge style. That url looks better than badgen.now.sh & flat-badgen.now.sh 🤔

BTW, As of (1), I even thought about shrink `https://badgen.now.sh/badge/...` to `https://badgen.now.sh/-/...` 😂 but that's not right. Do you have any better idea for shortening this `/badge/` segment?

**Comment by styfle** [comment]
Well the original badge is:

```
https://badgen.now.sh/badge/:subject/:status/:color
```

Why not use this for flat badges:

```
https://badgen.now.sh/flat/:subject/:status/:color
```

**Comment by styfle** [comment]
I guess my point is: the domain name should remain the same, regardless of the "style" of badge.

**Comment by amio** [comment]
That's ok for static badge, but a little complicated for live badges

**Comment by amio** [comment]
Segments in URL are like arguments for functions, I want to keep them as less as possible

**Comment by styfle** [comment]
I see what you mean. So changing the route isn't really an option.

-------

I think it's fine to add a query string parameter for badge style.
Since the data is the same so it make sense to keep the URL the same.

This is what shields does:

https://img.shields.io/npm/v/copee.svg
https://img.shields.io/npm/v/copee.svg?style=flat
https://img.shields.io/npm/v/copee.svg?style=flat-square
https://img.shields.io/npm/v/copee.svg?style=plastic

This also gives you flexibility if you want to add more options, no need to mess with the routes.

**Comment by amio** [comment]
I'm OK with query string, just not a fan of it. I was thinking if we could offer a better option than Shields does? My thoughts on this can be divided into two parts:

### Is there a better way for switching badge style?

What about put that info into hostname, compare:

    https://badgen.now.sh/npm/v/copee?style=flat

to

    https://flat-badgen.now.sh/npm/v/copee

I prefer the latter one.

This also brings an __extra benefit__, we could [easily](https://github.com/amio/badgen-service/blob/dd0aa0f19972a6fba6806cb991db556e1961de1c/libs/serve-badge.js#L4) serve a [full preview](https://flat.badgen.net) of all flat badges [within a same server]() instance:

	![](/npm/v/copee)

I like the simplicity.

And there's __another extra benefit__ 🤪: Put "flat" in the hostname makes it feels more "official" like, stand at the same level as "classic" style. After all, if someone want to use flat style badge, he would use it on all badges rather than in a mixed style, `https://flat-badgen.now.sh` will be the one-stop shop solely for those flat-lovers, they don't have to pick badge from classic shop, then "modify" it to flat style. I think it's a better experience.

I was think of this recently, to __provide consistent style across all badges__ could be a goal/value of Badgen. That's why Badgen use to redirect `travis-ci` badge requests to it's official location at first, but then turn to serve it as live-badge. These provider might not serve flat style, Badgen can do that. For the same purpose, first-class experience for flat-lovers is a meaningful feature.

That's why I tend to use hostname to switch style. If so, (the second part):

### What hostname looks better?

Here's some options came to me:

1.  ```
    https://badgen.now.sh
 	https://flat-badgen.now.sh
    ```
2.  ```
	https://badgen.now.sh
	https://flat.badgen.now.sh
	```
3.  ```
	https://badgen.net
	https://flat.badgen.net
	```

(3) looks best in this situation 😋

**Comment by styfle** [comment]
> we could easily serve a full preview of all flat badges within a single server

Yes that's a good point I didn't consider, but surely this could be accomplished with a query string parameter and maybe a dropdown to pick the style.

Take a look at https://shields.io/#styles which also shows additional options beyond `style` which might be beneficial such as `link` or `maxAge`.

> provide consistent style across all badges could be a goal/value of Badgen.

Yep, I agree, that's a good goal.

> (3) looks best in this situation

If I had to pick, 3 would be best.

One thing to note is that shields calls this style `flat-square` since `flat` is the default style. Maybe Badgen should use the same style name to avoid confusion?

**Comment by olstenlarck** [comment]
I'm agree too. In anyway there should have support for query params anyway. Because there we should allow passing `icon`, `emoji` and why not `label` (allows changing the left side to whatever you want, i really want that feature).

Lets have both `style` query and domain. The `badgen.net` will serve the flat rounded style and will allow to pass `style` query param. The `flat.badgen.net` will serve flat-squared style.

And actually i'm agree with @styfle for the naming. It may be confusing, but it would suck if we name it `flat-square.badgen.net` :D

**Comment by olstenlarck** [comment]
Just realizing that `find-my-way` router is pretty bullshit ;d
And its [query string tests](https://github.com/delvedor/find-my-way/blob/master/test/querystring.test.js) are ridicilous - testing nothing.

I just tried and we don't have way to get queryparams with this router.

I highly thinking to start a branch with `micro` and `micro-dev`

**Comment by amio** [comment]
- We should have query params support for sure 💪
- Let's give "style" switch(the most used switcher) a special treatment 😄
- Agree with the `flat`/`flat-square` naming confusion, here's some options with my thought(not sure yet):
  1. `flat-square.badgen.net` - too long, not cool anymore
  2. `s.badgen.net` - short but confusing
  3. `square.badgen.net` - ok maybe?
  4. `flat.badgen.net` - currently choice
  After rejecting option (1), consistency with shields.io is gone. Among other choices, flat is the shortest but still meaningful one, that's how I choose it. What's your idea about these options or other proposal?

As the `find-my-way` thing, here's my plan:
```
new URL('https://badgen.now.sh/badge?arg=1').searchParams
```

I like `micro` but it doesn't come with router, we still need to choose one between [these](https://github.com/amio/awesome-micro#modules).

**Comment by olstenlarck** [comment]
> I like micro but it doesn't come with router, we still need to choose one between these.

Exactly. Why not? Even more, we need just `micro-get` and `microrouter`.

As about the proposals.. i'm for `flat.badgen.net`. No matter that it may be confusing for the users - they'll get it.

**Comment by amio** [comment]
😂 Just thought the shape of fork kinda like paths of route. Maybe I should create a logo for it.

**Comment by amio** [comment]
https://flat.badgen.net online now 💃

**Comment by olstenlarck** [comment]
I think it would be better the "Query Params" to be on the top of the site, between "Usage" and the "Examples" sections. Because badges number will grow.

Also think that `style | flat` badge could be visually flat on the `badgen.net` and `style | classic` be visually classic on the `flat.badgen.net`.

**Comment by styfle** [comment]
> Also think that style | flat badge could be visually flat on the badgen.net and style | classic be visually classic on the flat.badgen.net.

@olstenlarck Fixed in #32

**Comment by amio** [comment]
> I think it would be better the "Query Params" to be on the top of the site, between "Usage" and the "Examples" sections. Because badges number will grow.

Agree 👍 At first I put that block on top, latter change to bottom for a little nicer(very opinionated) first screen. I'm planing move index.md to a fully customizable jsx page, with more refinement on the looks, I'll be happy to move "Query Params" to top.
